pkg/server/server.go

```diff
@@ -273,7 +273,14 @@ func New(ctx context.Context, config *lepconfig.Config, endpoint string, cliUID
 			allComponents = append(allComponents, c)
 
 		case os.Name:
-			allComponents = append(allComponents, os.New(ctx, rebootEventStore))
+			c, err := os.New(components.GPUdInstance{
+				RootCtx:          ctx,
+				RebootEventStore: rebootEventStore,
+			})
+			if err != nil {
+				return nil, fmt.Errorf("failed to create component %s: %w", k, err)
+			}
```

Comment:
we can init these components in a loop, so that we can remove all these switch.case eventually.

Reply:
Yes, that's the plan. I will do this in this PR

Reply:
Fixed all.